gradle-suppressible-error-prone/src/test/groovy/com/palantir/gradle/suppressibleerrorprone/SuppressibleErrorPronePluginIntegrationTest.groovy

```diff
@@ -1226,6 +1226,34 @@ class SuppressibleErrorPronePluginIntegrationTest extends IntegrationSpec {
         !rootCauseMessage.contains('[RemoveRolloutSuppressions]')
     }
 
+    def 'error-prone dependencies have versions bound together by a virtual platform'() {
+        setup: 'when an error-prone dependency is forced to certain version'
+        // language=Gradle
+        buildFile << '''
+            configurations.named('annotationProcessor') {
+                resolutionStrategy {
+                   force 'com.google.errorprone:error_prone_annotation:2.3.4'
+                }
+            }
+            
+            tasks.register('printErrorProneVersions') {
+                inputs.files(configurations.named('annotationProcessor'))
+                doLast {
+                    inputs.files.files.each {
+                        println("ERROR-PRONE: ${it.name}")
+                    }
+                }
+            }
+        '''.stripIndent(true)
+
+        when:
+        def output = runTasksSuccessfully('printErrorProneVersions').standardOutput
+
+        then: 'every single error-prone dependency has the same version'
+        output.contains('ERROR-PRONE: error_prone_annotation-2.3.4.jar')
+        output.contains('ERROR-PRONE: error_prone_core-2.3.4.jar')
+    }
```

Comment:
what would happen if you didn't have a force? is there a way to do a before and after style test? Or to recreate the scenario you're talking about

Reply:
I don't think I can, without adding a flag or something to similar to disable the ComponentMetadataRule that I've added. Now these versions will always be the same, I can't replicate the previous issue (as it's fixed).